File: javascript/src/televentas/pedido.js
export class Pedido {
  /**
   * @param {number} idPedido
   * @param {string} estado
   * @param {Date} fechaPreparacion
   * @param {boolean} empaquetado
   * @param {import('./ordenCompra.js').OrdenCompra} orden
   */
  constructor(idPedido, estado, fechaPreparacion, empaquetado, orden) {
    this.idPedido = idPedido;
    this.estado = estado;
    this.fechaPreparacion = fechaPreparacion;
    this.empaquetado = empaquetado;
    this.orden = orden;
  }

  getIdPedido() {
    return this.idPedido;
  }

  getEstado() {
    return this.estado;
  }

  getFechaPreparacion() {
    return this.fechaPreparacion;
  }

  isEmpaquetado() {
    return this.empaquetado;
  }

  getOrden() {
    return this.orden;
  }

  preparar() {
    this.estado = "en_preparacion";
  }

  empaquetar() {
    this.empaquetado = true;
    this.estado = "empaquetado";
  }

  marcarListo() {
    if (!this.empaquetado) {
      throw new Error("No se puede marcar como listo un pedido no empaquetado.");
    }
    this.estado = "listo";
  }
}

File: java/src/televentas/Pedido.java
package televentas;

import java.time.LocalDate;

public class Pedido {

    private final int idPedido;
    private String estado;
    private final LocalDate fechaPreparacion;
    private boolean empaquetado;
    private final OrdenCompra orden;

    public Pedido(int idPedido, String estado, LocalDate fechaPreparacion, boolean empaquetado, OrdenCompra orden) {
        this.idPedido = idPedido;
        this.estado = estado;
        this.fechaPreparacion = fechaPreparacion;
        this.empaquetado = empaquetado;
        this.orden = orden;
    }

    public int getIdPedido() {
        return idPedido;
    }

    public String getEstado() {
        return estado;
    }

    public LocalDate getFechaPreparacion() {
        return fechaPreparacion;
    }

    public boolean isEmpaquetado() {
        return empaquetado;
    }

    public OrdenCompra getOrden() {
        return orden;
    }

    public void preparar() {
        this.estado = "en_preparacion";
    }

    public void empaquetar() {
        this.empaquetado = true;
        this.estado = "empaquetado";
    }

    public void marcarListo() {
        if (!empaquetado) {
            throw new IllegalStateException("No se puede marcar como listo un pedido no empaquetado.");
        }
        this.estado = "listo";
    }
}
